contracts/serverless-v2/Executors.sol

```diff
@@ -0,0 +1,470 @@
+import "../interfaces/IAttestationVerifier.sol";
+
+contract Executors is
+    Initializable, // initializer
+    ContextUpgradeable, // _msgSender, _msgData
+    ERC165Upgradeable, // supportsInterface
+    AccessControlUpgradeable, 
+    UUPSUpgradeable, // public upgrade
+    AttestationAutherUpgradeable,
+    TreeUpgradeable
+{
+    using SafeERC20 for IERC20;
+    using ECDSA for bytes32;
+
+    error ExecutorsZeroAddressToken();
+    error ExecutorsZeroMinStakeAmount();
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    // initializes the logic contract without any admins
+    // safeguard against takeover of the logic contract
+    constructor(
+        IAttestationVerifier attestationVerifier,
+        uint256 maxAge,
+        IERC20 _token,
+        uint256 _minStakeAmount
+    ) AttestationAutherUpgradeable(attestationVerifier, maxAge) {
+        _disableInitializers();
+
+        if(address(_token) == address(0))
+            revert ExecutorsZeroAddressToken();
+        if(_minStakeAmount == 0)
+            revert ExecutorsZeroMinStakeAmount();
+
+        TOKEN = _token;
+        MIN_STAKE_AMOUNT = _minStakeAmount;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    function supportsInterface(
+        bytes4 interfaceId
+    )
+        public
+        view
+        virtual
+        override(ERC165Upgradeable, AccessControlUpgradeable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function _authorizeUpgrade(
+        address /*account*/
+    ) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    error ExecutorsZeroAddressAdmin();
+
+    function initialize(
+        address _admin,
+        EnclaveImage[] memory _images
+    ) public initializer {
+        if(_admin == address(0))
+            revert ExecutorsZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+        __AttestationAuther_init_unchained(_images);
+        __TreeUpgradeable_init_unchained();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable TOKEN;
+    // TODO: add min stake limit and if it falls below that limit then remove from tree
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable MIN_STAKE_AMOUNT;
+    Jobs public jobs;
+
+    bytes32 public constant JOBS_ROLE = keccak256("JOBS_ROLE");
+
+    function setJobsContract(Jobs _jobs) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
as in, the admin can just use `grantRole`/`revokeRole` to manage the jobs contract

don't need this or the jobs variable